Automattic/wp-calypso
# Segmentation Survey: Move queries and mutations to client/data

Closes https://github.com/Automattic/wp-calypso/issues/89500

## Proposed Changes

* Move queries and mutations to client/data
* Create tests for `useCachedAnswers` and `useSaveAnswersMutation`

## Testing Instructions

* Apply this PR to your local
* Run tests locally `yarn test-client client/data/segmentaton-survey`
* Revert the last commit to access the testing route `git revert 5e1f589`
* Go to http://calypso.localhost:3000/ecommerce-segmentation-survey?flags=ecommerce-segmentation-survey
* Verify if the Survey structure is loaded and rendered as expected
* Choose an option and click Continue
* Verify on Network tab if the save endpoint was called as expected

## Pre-merge Checklist
